akka-http/src/main/scala/akka/http/javadsl/server/directives/RouteDirectives.scala

```diff
@@ -35,6 +34,21 @@ abstract class RouteDirectives extends RespondWithDirectives {
   private implicit val conversionExecutionContext = ExecutionContexts.sameThreadExecutionContext
 
   /**
+   * Partially applying functions for the Java World!
+   * @param f the function to partially apply
+   * @param a the first parameter to partially apply
+   * @tparam A the type of the applied parameter
+   * @tparam B the type of the second parameter
+   * @tparam R the type of the return
+   * @return the function partially applied
+   */
+  def partial[A, B, R](f: BiFunction[A, B, R], a: A): JFunction[B, R] = {
```

Comment:
Since this is not a directive, and not even http-specific, it should probably go to a different class/package.

It also should be properly documented, since the average Java developer wouldn't know what "partially applying" a function is. Actually, isn't the act of partially applying called [_currying_](https://en.wikipedia.org/wiki/Currying) or am I mixing sth up? In that case, `curry` would be a more apt method name.